controllers/machine_controller.go

```diff
@@ -95,6 +96,14 @@ func (r *MachineReconciler) SetupWithManager(mgr ctrl.Manager, options controlle
 		return errors.Wrap(err, "failed setting up with a controller manager")
 	}
 
+	// Add index to Machine for listing by Node reference
+	if err := mgr.GetCache().IndexField(&clusterv1.Machine{},
+		machineNodeNameIndex,
+		r.indexMachineByNodeName,
```

Comment:
I'm not really sure we can get the same indexer with two different indexerFunc 
From a quick peek at the code, it seems that the last indexerFunc takes over, so I'm wondering if we should move the index creation in the main, right after the tracker creation, instead of having this both in MHC and here
@vincepri ^^

Reply:
I moved the indexer creation to Machine controller, removed from MHC. It was not allowed to call indexer to same field twice. Both MHC and Machine needs same indexerFunc so Machine controller is added first in the main, hence moved it there.

Reply:
Let's maybe add it to the main.go, so it's clear it's generic